lib/ace/mode/vbscript_highlight_rules.js

```diff
@@ -41,36 +41,42 @@ var TextHighlightRules = require("./text_highlight_rules").TextHighlightRules;
 var VBScriptHighlightRules = function() {
 
     var keywordMapper = this.createKeywordMapper({
-        "keyword.control.asp":  "If|Then|Else|ElseIf|End|While|Wend|For|To|Each|Case|Select|Return"
-            + "|Continue|Do|Until|Loop|Next|With|Exit|Function|Property|Type|Enum|Sub|IIf|Class",
-        "storage.type.asp": "Dim|Call|Const|Redim|Set|Let|Get|New|Randomize|Option|Explicit",
```

Comment:
`return` and `continue` seem to be keywords in Visual Basic

https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/return-statement
https://docs.microsoft.com/en-us/dotnet/visual-basic/language-reference/statements/continue-statement

`iif` is also highlighted as keyword in other editors https://docs.microsoft.com/en-us/office/vba/language/reference/user-interface-help/iif-function

Reply:
Isn't this VBscript mode? 

VBScript, Visual Basic, VB.NET and Visual Basic for Applications (VBA) are complete different environments. 

[VBScript Language Reference](https://docs.microsoft.com/en-us/previous-versions/windows/internet-explorer/ie-developer/scripting-articles/d1wf56tt(v=vs.84)) is the official reference.

Reply:
BTW the link you referenced for `iif` is from Visual Basic for Applications,
the others for `return` and `continue` are from Visual Basic .NET documentation.

Reply:
This was meant to be used with all four versions of the language, similar to https://github.com/SublimeText/VBScript.

So if these versions are sufficiently different to require different modes, we will need to make the mode configurable.

Are there more differences between these four, and do they use different file extensions?

Reply:
> This was meant to be used with all four versions of the language, similar to https://github.com/SublimeText/VBScript.

Then it's a very bad name choice you followed. `VB` maybe covers semantically all those four languages just for syntax highlighting, but `VBScript` points a precise environment and language, the most restricted language of the group, that's why I made this PR.

If you trust my 20 years of experience in Visual Basic, there's two good way to do this in logical manner.

1. Three different mode or configuration
    1. `vb`: Visual Basic (includes Visual Basic for Applications aka VBA)
    2. `vbscript`: VBScript
    3. `vbnet`: Visual Basic .NET
2. One mode called `vb` that tries to support all those languages.


> Are there more differences between these four, and do they use different file extensions?

There are much differences in syntax and yes they use different file extensions.

#### Visual Basic (VB Classic) and Visual Basic for Applications (VBA):

- A programming language.
- Requires its own runtime or MS Office.
- Same language, different environments. VBA is still being updated, VB Classic is obsolete.
- **Primary Extensions:** .cls, .bas
		
#### VBScript:

- A scripting language.
- Requires Windows OS. Preinstalled since Windows 2000.
- It's a subset of VB Classic and VBA.
- Almost any code written in VBScript also works in VB Classic and VBA, but not the other way around.
- **Primary Extensions:** .vbs
- **Other Extensions:** .wsc, .hta, .html, .asp, .asa

	
#### Visual Basic .NET:

- A programming language supported with .NET
- A completely different language from the other three languages.
- Up to date.
- Provides some very limited backward compatibility for VB Classic.
- **Primary Extensions:** .vb
- **Other Extensions:** .aspx, .vbhtml, .ashx, .asax

In the end, Ace is a very mature project. The decision is yours.

Reply:
Thanks for the explanation. Considering that they have different file extensions, different modes seems to be a better solution.
But if we were to create three different modes,  how different the highlight rules would be. 
Would it be possible to use one file with options object similar to https://github.com/ajaxorg/ace/blob/master/lib/ace/mode/javascript_highlight_rules.js#L359 ?

Reply:
I'm not familiar with Ace's API to create or understand such rules.
But if creating independent rule sets is an option I definitely can help with that.

Reply:
Having independent rule sets would be very useful. And if they end up similar enough i can merge them to optimize the downloaded file size, as it wouldn't require knowledge of the vb language.

Reply:
Then I'm closing this pull request and see what I can do. I'll let you know.